81-question-mark-chrysalis/technical_specification.md

```diff
@@ -0,0 +1,230 @@
+
+`EMPack`: Experimental metadata in datapack format. It follows a relation data schema and compatible with one of the EMIMs.
+
+`AnnotatedEMPack`: A datapack enriched with additional information. 
+
+`AnnotatedEMPacks`: A MongoDB collection of AnnotatedEMPack objects.
+
+`Workflow`: Instructions on how to produce a certain data/model representation from another data/model. The workflows are defined by the `metldata` library on which the transformation service will be built.
+
+`Workflows`: A MongoDB collection of Workflow objects.
+
+`WorkflowRoute`: Information on which specific workflow is used to produce data compliant with the output model when presented with data compliant with the input model.
+
+`WorkflowRoutes`: A MongoDB collection of WorkflowRoute objects.
+
+
+### Included/Required:
+
+- Implement logic to process incoming AnnotatedEMPacks, execute the corresponding workflows and store the resulting AnnotatedEMPacks if required.
+- Implement service start-up logic to derive missing schemas for workflow routes with empty output schemas.
+- Implement MongoDB collections to store AnnotatedEMPacks, Workflows and WorkflowRoutes.
+
+#### Collections
+
+
+`Schemas`
+
+- purpose: store schemas
+
+data structure:
+```python
+class Schema(BaseModel):
+   id: uuid
+   name: str
+   original: bool
+   version: str | None
+   content: SchemaPack | None
+   publish: bool 
+   order: int | None
+
+```
+
+`Workflows`
+
+- Purpose: store metldata-compatible workflow definitions used to transform schemas/data.
+
+data structure:
+```python
+class TransformationWorkflow(BaseModel):
+   id: uuid
+   name: str | None
+   description: str | None
+   workflow: metldata.workflow.base.Workflow
+```
+
+
+`WorkflowRoutes`
+
+- Purpose: describe named routes that bind workflows, input schema types and produced output schema types, and whether outputs should be published.
+
+data structure:
+```python
+class WorkflowRoute(BaseModel):
+   id: uuid
+   name: str | None
+   input_schema_id: uuid
+   output_schema_id: uuid
+   workflow_id: uuid
+```
+
+
+`AnnotatedEMPacks`
+
+Purpose: store incoming and derived annotated EM datapacks that are to be processed or published.
+
+data structure:
+```python
+class AnnotatedEMPack(BaseModel):
+   id: uuid
+   schema_id: uuid
+   original_id: uuid | None
+   datapack: DataPack
+   annotation: dict
+```
+
+#### Implementation Details of Collections
+
+The implementation of the EM transformation service will involve the following key components.
+
+1. `Schemas`
+   
+   Consists of Schema objects
+   * `id` is a unique identifier of the schema
+   * `name` is a human readable name of the schema
+   * `original` is a boolean flag indicating whether the schema is an original or derived schema
+   * `version` is the version of the schema (aka type of the schema), initially None if it is not an original schema
```

Comment:
I would not use the term type here, in my view this should be just the version number (like "1.0"), and the type should be reflected by the name.